include/RAJA/policy/cuda/raja_cudaerrchk.hpp

```diff
@@ -43,33 +109,95 @@ namespace RAJA
+                       std::forward_as_tuple(__VA_ARGS__),                \
+                       __FILE__, __LINE__);                               \
   }
 
-inline void cudaAssert(cudaError_t code,
-                       const char* file,
-                       int line,
-                       bool abort = true)
+// returns a space separated string of the arguments to the given function
+// returns an empty string if func is unknown
+// no leading or trailing spaces
+constexpr std::string_view cuda_api_arg_names(std::string_view func)
+{
+  using storage_type = std::pair<std::string_view, std::string_view>;
+  constexpr std::array<storage_type, 18> known_functions{{
+    storage_type{"cudaDeviceSynchronize",                         ""},
+    storage_type{"cudaGetDevice",                                 "device"},
+    storage_type{"cudaGetDeviceProperties",                       "prop device"},
+    storage_type{"cudaStreamSynchronize",                         "stream"},
+    storage_type{"cudaHostAlloc",                                 "pHost size flags"},
+    storage_type{"cudaHostFree",                                  "ptr"},
+    storage_type{"cudaMalloc",                                    "devPtr size"},
+    storage_type{"cudaFree",                                      "devPtr"},
+    storage_type{"cudaMemset",                                    "devPtr value count"},
+    storage_type{"cudaMemcpy",                                    "dst src count kind"},
+    storage_type{"cudaMemsetAsync",                               "devPtr value count stream"},
+    storage_type{"cudaMemcpyAsync",                               "dst src count kind stream"},
+    storage_type{"cudaLaunchKernel",                              "func gridDim blockDim args sharedMem stream"},
+    storage_type{"cudaPeekAtLastError",                           ""},
+    storage_type{"cudaGetLastError",                              ""},
+    storage_type{"cudaFuncGetAttributes",                         "attr func"},
+    storage_type{"cudaOccupancyMaxPotentialBlockSize",            "minGridSize blockSize func dynamicSMemSize blockSizeLimit"},
+    storage_type{"cudaOccupancyMaxActiveBlocksPerMultiprocessor", "numBlocks func blockSize dynamicSMemSize"}
+  }};
+  for (auto [api_name, api_args] : known_functions) {
+    if (func == api_name) {
+      return api_args;
+    }
+  }
+  return "";
+}
+
+template < typename Tuple >
+RAJA_INLINE void cudaAssert(cudaError_t code,
+                            std::string_view func_name,
+                            std::string_view args_name,
+                            Tuple const& args,
+                            std::string_view file,
+                            int line,
+                            bool abort = true)
 {
   if (code != cudaSuccess)
   {
+    std::ostringstream str;
+    str << "CUDAassert: ";
```

Comment:
I think it would be more readable as "CUDA assert:" or "CUDA error:" (with a space after CUDA).

Reply:
I think I like "error" better than "assert" for what will be a user facing message.